core/node/state_keeper/src/keeper.rs

```diff
@@ -49,28 +101,60 @@ pub(super) const POLL_WAIT_DURATION: Duration = Duration::from_secs(1);
 /// You can think of it as a state machine that runs over a sequence of incoming transactions, turning them into
 /// a sequence of executed L2 blocks and batches.
 #[derive(Debug)]
-pub struct ZkSyncStateKeeper {
+pub struct StateKeeper {
+    inner: StateKeeperInner,
+    batch_state: BatchState,
+}
+
+#[derive(Debug)]
+pub struct StateKeeperBuilder {
     io: Box<dyn StateKeeperIO>,
     output_handler: OutputHandler,
-    batch_executor: Box<dyn BatchExecutorFactory<OwnedStorage>>,
+    batch_executor_factory: Box<dyn BatchExecutorFactory<OwnedStorage>>,
     sealer: Arc<dyn ConditionalSealer>,
     storage_factory: Arc<dyn ReadStorageFactory>,
     health_updater: HealthUpdater,
     deployment_tx_filter: Option<DeploymentTxFilter>,
 }
 
-impl ZkSyncStateKeeper {
+/// Helper struct that encapsulates some private state keeper methods.
+#[derive(Debug)]
+pub(super) struct StateKeeperInner {
```

Comment:
This has the same fields as `StateKeeperBuilder`. Is this intentional? Would it make sense to encapsulate `_Inner` in `_Builder` (if the goal is to separate them by the usage pattern) or use a single type (i.e., separate by method visibility)?

Reply:
IMO it's cleaner to keep separate types. It adds a little code and usage pattern is straightforward this way. Also, in future it's possible they will have different set of fields